server.go

```diff
@@ -235,27 +244,35 @@ func (sc *serverConn) writeReset(strm uint32, code ErrorCode) {
 	sc.writer <- fr
 }
 
-func (sc *serverConn) writeGoAway(strm uint32, code ErrorCode) {
+func (sc *serverConn) writeGoAway(strm uint32, code ErrorCode, message string) {
 	ga := AcquireFrame(FrameGoAway).(*GoAway)
 
 	fr := AcquireFrameHeader()
-	fr.SetStream(strm)
 
 	ga.SetStream(strm)
 	ga.SetCode(code)
+	ga.SetData([]byte(message))
 
 	fr.SetBody(ga)
 
 	sc.writer <- fr
 }
 
 func (sc *serverConn) writeError(strm *Stream, err error) {
-	code := InternalError
-	if errors.Is(err, Error{}) {
-		code = err.(Error).Code()
+	streamErr := Error{}
+	if !errors.As(err, &streamErr) {
```

Comment:
If you are checking the opposite of the error, why not using `!errors.Is(err, Error{})`?

Reply:
Not sure to understand, I use the streamErr after the if so I keep the cast in some way.

Reply:
That's right. Sorry, didn't notice that.